azurerm/resource_arm_netapp_pool.go

```diff
@@ -0,0 +1,220 @@
+	accountName := id.Path["netAppAccounts"]
+	name := id.Path["capacityPools"]
+
+	resp, err := client.Get(ctx, resourceGroup, accountName, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] NetApp Pools %q does not exist - removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error reading NetApp Pools %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	d.Set("name", name)
+	d.Set("resource_group_name", resourceGroup)
+	d.Set("account_name", accountName)
+	if location := resp.Location; location != nil {
+		d.Set("location", azure.NormalizeLocation(*location))
+	}
+	if poolProperties := resp.PoolProperties; poolProperties != nil {
+		if err := d.Set("service_level", poolProperties.ServiceLevel); err != nil {
+			return fmt.Errorf("Error setting `service_level`: %+v", err)
+		}
+
+		if err := d.Set("size", poolProperties.Size); err != nil {
+			return fmt.Errorf("Error setting `size`: %+v", err)
+		}
+	}
+
+	return nil
+}
+
+func resourceArmNetAppPoolDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).Netapp.PoolClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := azure.ParseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	accountName := id.Path["netAppAccounts"]
+	name := id.Path["capacityPools"]
+
+	_, err = client.Delete(ctx, resourceGroup, accountName, name)
+	if err != nil {
+		return fmt.Errorf("Error deleting NetApp Pool %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	return waitForNetAppPoolToBeDeleted(ctx, client, resourceGroup, accountName, name)
+}
+
+func waitForNetAppPoolToBeDeleted(ctx context.Context, client *netapp.PoolsClient, resourceGroup, accountName, name string) error {
+	log.Printf("[DEBUG] Waiting for NetApp Pool Provisioning Service %q (Resource Group %q) to be deleted", name, resourceGroup)
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{"200", "202"},
+		Target:  []string{"404"},
+		Refresh: netappPoolDeleteStateRefreshFunc(ctx, client, resourceGroup, accountName, name),
+		Timeout: 20 * time.Minute,
+	}
+	if _, err := stateConf.WaitForState(); err != nil {
+		return fmt.Errorf("Error waiting for NetApp Pool Provisioning Service %q (Resource Group %q) to be deleted: %+v", name, resourceGroup, err)
+	}
+
+	return nil
+}
+
+func netappPoolDeleteStateRefreshFunc(ctx context.Context, client *netapp.PoolsClient, resourceGroupName string, accountName string, name string) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		res, err := client.Get(ctx, resourceGroupName, accountName, name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(res.Response) {
+				return nil, "", fmt.Errorf("Error retrieving NetApp Pool %q (Resource Group %q): %s", name, resourceGroupName, err)
+			}
+		}
+
+		if _, err := client.Delete(ctx, resourceGroupName, accountName, name); err != nil {
```

Comment:
If the future isn't quite accurate, do you mind commenting about that and also comment about why we're deleting again here

Reply:
Also opening an issue on the SDK with a link here so we can track when its fixed

Reply:
Sure. Updated.

Reply:
Looks great!